src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4853,9 +4857,19 @@ bool QgsWithVariableExpressionFunction::prepare( const QgsExpressionNodeFunction
   QVariant name = args->at( 0 )->prepare( parent, context );
   QVariant value = args->at( 1 )->prepare( parent, context );
 
-  appendTemporaryVariable( context, name.toString(), value );
-  args->at( 2 )->prepare( parent, context );
-  popTemporaryVariable( context );
+  QgsExpressionContext *updatedContext = const_cast<QgsExpressionContext *>( context );
```

Comment:
Is this cast required? Isn't the one in `appendTemporaryVariable` enough?

Reply:
You're right